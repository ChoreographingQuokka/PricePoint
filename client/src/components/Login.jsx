import React from 'react';
import ReactDOM from 'react-dom';
import { Grid, Row, Col } from 'react-bootstrap';
import {
  Button, Glyphicon, PageHeader,
  Form, FormGroup, ControlLabel, FormControl,
} from 'react-bootstrap';
import { Link } from 'react-router-dom';
import Header from './Header.jsx';

const Login = () => {
  return (
    <div>
      <Grid>
        <Row>
          <Col sm={6} smOffset={3}>
            <h1>
              <span>
                Login
              </span>
            </h1>
            {/* some conditional rendering logic:
            if (message.length) { %>
            <div class="alert alert-danger"><%= message %></div>
            <% } */}
            <Form action="/login" method="post">
              <FormGroup>
                <ControlLabel>
                  Email
                </ControlLabel>
                <FormControl
                  className="signup-input"
                  type="text"
                  name="email"
                />
                <FormControl.Feedback />
              </FormGroup>
              <FormGroup>
                <ControlLabel>
                  Password
                </ControlLabel>
                <FormControl
                  className="signup-input"
                  type="password"
                  name="password"
                />
                <FormControl.Feedback />
              </FormGroup>
              <Button type="submit" bsSize="large" bsStyle="warning">Login</Button>
            </Form>
            <div>
              <PageHeader></PageHeader>
              Or login with any of the following services:
              <br/>
              <a href="/auth/facebook">
                <img className="fblogo" src="/assets/facebook.svg"/>
              </a>
              <a href="/auth/google">
                <img className="googlogo" src="/assets/google.svg"/>
              </a>
              <a href="/auth/twitter">
                <img className="twitlogo" src="/assets/twitter.svg"/>
              </a>
            </div>
            <PageHeader></PageHeader>
            <div>
              <p>
                Need to sign up for an account?
                <Link to="/signup"> Signup</Link>
              </p>
              <p>
                <Link to="/">home</Link>
              </p>
            </div>
          </Col>
        </Row>
      </Grid>
    </div>
  );
};

export default Login;
