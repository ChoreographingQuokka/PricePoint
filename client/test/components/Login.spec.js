import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Login from '../../src/components/Login';

describe('<Login />', () => {
  xit('renders', () => {
    const wrapper = shallow(<Login />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
