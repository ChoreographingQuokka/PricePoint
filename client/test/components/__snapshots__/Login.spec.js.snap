// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders 1`] = `
<div>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Header />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        sm={6}
        smOffset={3}
      >
        <h1>
          <span>
            <Glyphicon
              bsClass="glyphicon"
              bsSize="small"
              glyph="log-in"
            />
             Login
          </span>
        </h1>
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <FormGroup
            bsClass="form-group"
            controlId="formBasicText"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Email
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange=""
              type="text"
              value=""
            />
            <FormControlFeedback
              bsClass="form-control-feedback"
              bsRole="feedback"
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="formBasicText"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Password
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange=""
              type="text"
              value=""
            />
            <FormControlFeedback
              bsClass="form-control-feedback"
              bsRole="feedback"
            />
          </FormGroup>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="warning"
            disabled={false}
          >
            Login
          </Button>
        </Form>
        <div>
          <PageHeader
            bsClass="page-header"
          />
          Or login with any of the following services:
          <br />
          <a
            href="/auth/facebook"
          >
            <img
              src="/assets/fb-logo.png"
              style={
                Object {
                  "height": "29px",
                  "marginRight": "15px",
                  "width": "29px",
                }
              }
            />
          </a>
          <a
            href="/auth/google"
          >
            <img
              src="/assets/google-logo.png"
              style={
                Object {
                  "height": "29px",
                  "marginLeft": "15px",
                  "marginRight": "15px",
                  "width": "29px",
                }
              }
            />
          </a>
          <a
            href="/auth/twitter"
          >
            <img
              src="/assets/twitter-logo.png"
              style={
                Object {
                  "height": "48px",
                  "marginLeft": "15px",
                  "width": "48px",
                }
              }
            />
          </a>
        </div>
        <PageHeader
          bsClass="page-header"
        />
        <div>
          <p>
            Need to sign up for an account?
            <Link
              replace={false}
              to="/signup"
            >
               Signup
            </Link>
          </p>
          <p>
            <Link
              replace={false}
              to="/"
            >
              home
            </Link>
          </p>
        </div>
      </Col>
    </Row>
  </Grid>
</div>
`;
